hedera-node/src/main/java/com/hedera/services/context/TransfersHelper.java

```diff
@@ -0,0 +1,86 @@
+package com.hedera.services.context;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.util.Arrays;
+
+public class TransfersHelper {
+	private TransfersHelper() {
+		throw new UnsupportedOperationException("Utility class");
+	}
+
+	public static Triple<long[], long[], Integer> updateFungibleChanges(
```

Comment:
I think we can actually do a lot better by allocating the parallel `long[]` arrays just _once_ in `SideEffectsTracker` and updating them **in place**. (Then all we need to do in `SideEffectsTracker.reset()` is reset a `int touchedSoFar` to zero.) 

On the `fungible-change-mgmt-idea` branch I have a candidate implementation [here](https://github.com/hashgraph/hedera-services/blob/fungible-change-mgmt-idea/hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java#L569) for the update method; and [here](https://github.com/hashgraph/hedera-services/blob/fungible-change-mgmt-idea/hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java#L609) for the zero-purge method---and a unit test [here](https://github.com/hashgraph/hedera-services/blob/fungible-change-mgmt-idea/hedera-node/src/test/java/com/hedera/services/context/SideEffectsTrackerTest.java#L306) that validates 1M randomly constructed `TransferList`s w/ the results of the current implementation. 

Am curious what kind of throughput that will do... 🤔

Reply:
Thanks Michael ! With the existing change it is 2x faster and with updating in place as you suggested it is 5X faster. 
These are the numbers I see from the benchmark, on average
1. Using TransferList  `102379.684 ops/s`
2.  using List<Long> in SideEffectsTracker `168440.572 ops/s`  
3. using long[] array in SideEffectsTracker `254957.561` and copies
4. using in place array modification `505066 ops/s`

Reply:
That's awesome @Neeharika-Sompalli! Tyty